### PR TITLE
pick.lic - Added args to ignore exp mindstate limits and skip perception checks

### PR DESCRIPTION
**Pull request by mikejcook**
Two new arguments, "force" and "perception"

Force: ignore experience mindstate limits that would normally stop picking
Perception: Skip identify checks (for when perception skill is insufficient)

### CODE REVIEW

File: pick.lic [hunk 1]
```diff
@@ -116,12 +116,16 @@ class LockPicker
...
+        { name: 'force', regex: /force/i, optional: true, description: 'Force picking, ignoring any experience mindstate limits' },
+        { name: 'perception', regex: /perception/i, optional: true, description: 'Skip perception checks' }
       ]
     ]
 
     args = parse_args(arg_definitions, true)
-
+	
+    @skip_identify = args.perception
+    @force_repair = args.force
```
Comment (rpherbig) [inline pick.lic:128]: `_repair` doesn't feel like the right name. What does this have to do with repairing?

Comment (mikejcook) [reply pick.lic]: Corrected with new commit

File: pick.lic [hunk 2]
```diff
@@ -68,7 +68,7 @@ class LockPicker
     @harvest_traps = false
 
     boxes.each do |box|
-      if DRSkill.getxp('Locksmithing') >= 30
+      if DRSkill.getxp('Locksmithing') >= 30 && !force_picking
```
Comment (rpherbig) [inline pick.lic:71]: Should this be `@force_picking`?

File: pick.lic [hunk 3]
```diff
@@ -116,12 +116,16 @@ class LockPicker
         { name: 'pets', regex: /pets/, optional: true, description: 'Disarm boxes and place them in the pet box container.' },
         { name: 'count', regex: /\d+/, optional: true, description: 'How many pet boxes to make.' },
         { name: 'refill', regex: /refill/, optional: true, description: 'Refills your lockpick ring.' },
-        { name: 'source', regex: /(source=(\w+))/, optional: true, description: 'Container with boxes to pick.'}
+        { name: 'source', regex: /(source=(\w+))/, optional: true, description: 'Container with boxes to pick.'},
+        { name: 'force', regex: /force/i, optional: true, description: 'Force picking, ignoring any experience mindstate limits' },
```
Comment (rpherbig) [inline pick.lic:120]: How does this arg differ from setting `stop_pick_on_mindlock` to false?

Comment (mikejcook) [reply pick.lic:120]: This can be done on the fly, rather than having to change your YAML. I generally prefer to stop on mindlock, but if I'm e.g. opening boxes for someone else, that isn't the case.

Comment (Dartellum) [reply pick.lic:120]: It is not different. If I set `stop_pick_on_mindlock: false` I keep picking when I hit 34/34.

**Comment by Dartellum** [comment]
**FYI** A very large rewrite of pick is underway, see PR https://github.com/rpherbig/dr-scripts/pull/5001.

**Comment by mikejcook** [comment]
> 
> 
> **FYI** A very large rewrite of pick is underway, see PR #5001.

I've gone ahead and closed the pull request, since there's a re-write already, but you might consider adding one or both args to the new re-write. I use them in my own local version of pick. Yes, you can have a different YAML, but it's nice being able to just throw "force" to keep picking past locked. And then, some locks I am not able to identify but can disarm/pick, and the script just hangs forever on trying, hence why I added that argument.

**Comment by rpherbig** [comment]
I can see the use case for wanting a script arg in addition to a YAML setting. The only thing I'd suggest is using the script arg to modify the assignment to `stop_on_mindlock`, instead of adding a new conditional
